05-jupyter-ecosystem.md

```diff
@@ -336,8 +336,51 @@ time to complete.
 
 ### Explaining each cell
 
-A good rule of thumb is to always include a markdown cell above every code cell.
-
+A good rule of thumb is to always include a markdown cell above every
+code cell. In fact, instead of adding comments in code, consider
+moving the comments out of the code cell into a markdown cell either
+directly above or below the code cell. This often reads much better,
+and gives you more flexibilty in discussing or describing the
+code. However, short comments in a block of code can still be useful.
+
+
+### How much code should go in a cell?
+
+How much code should you put in a cell? You will develop your own
+style of writing noteooks with experience, but we recommend putting
+code that "goes together as a meaningful unit" into a single cell. For
+example, if you have lines of code that are highly dependent on each
+other, then you might want to put them together. As an example,
+consider two lines of code: one that creates an item, and the second
+that initializes it. It may be a good idea to put those into the same
```

Comment:
I'm not sure this will resonate with people who follow RAII. Perhaps use the example of creating and plotting (or computing statistics of) multiple related quantities, where a mismatch would produce an error or mislabeled/misleading plot?

Reply:
Perhaps a simpler example would be: opening and reading data from a file. Opening the file in one cell and reading in the data in another cell could introduce unexpected behavior.

Reply:
Yes, that sounds better! Thanks!